### ISSUE TITLE
Any ways to set font features.

### ISSUE THREAD

**Issue by iRealistiX** [cog1to/st-ligatures#16]
The title is clear, the patch is just **amazing**, especially since I love having lignatures, + the fact I am switching to neovim.
One thing I want to ask is how can I set font features (ss01, calt, etc) because the font I use needs them for ligs to render properly

**Comment by cog1to** [comment]
hello. thanks for the feedback. I didn't know about font features, but on the surface there seems to be a relatively easy way to add support for these via st's config. I'll try to work on that and update the patches

can you give me a font name or a link to the font that has this issue so I could test it?

**Comment by iRealistiX** [comment]
The fonts I tested with
`Cascadia Code Italic` with the calt features (required to display cursive italics)
`Fira Code` and the infinties arrows

**Comment by cog1to** [comment]
hm, I got Cascadia Code Italic, but it seems to work fine on my end:

![cascadia-ligas](https://user-images.githubusercontent.com/1484965/129970474-1d753ceb-f2e9-40c9-9182-2fbae81c661e.png)

maybe there are some other patches interfering with the shaping/rendering. have you applied any other patches?
another possibility is some specific program messing it up...

**Comment by iRealistiX** [comment]
I have font2, scrollback, disable bold italics (these are the ones that may interfer)
I will try a clean install with only the patch installed.

**Comment by cog1to** [comment]
@iRealistiX interesting. this might be a similar issue to the one I fixed earlier with boxdraw. I'll look into that more closely. Can you post your `font2` config value here?

**Comment by iRealistiX** [comment]
> @iRealistiX interesting. this might be a similar issue to the one I fixed earlier with boxdraw. I'll look into that more closely. Can you post the `font2` config value you had?

It's not the font2 patch problem, it was me who (by hand) patched it the wrong way for another patch that is disable bold italic

TL;DR: I did a bad job at patching

**Comment by cog1to** [comment]
@iRealistiX Add a PR with font features support. It's a little clunky: you'll need to add them to the `features` array in `hb.c`.

Would be cool if you can test it out before merge.
